discovery-provider/src/tasks/entity_manager/entity_manager.py

```diff
@@ -250,7 +271,73 @@ def fetch_existing_entities(
         )
         .all()
     )
-    existing_entities["users"] = {user.user_id: user for user in users}
+    existing_entities[EntityType.USER] = {user.user_id: user for user in users}
+
+    # FOLLOWS
+    follow_ops_to_fetch: Set[Tuple] = entities_to_fetch[EntityType.FOLLOW]
+    and_queries = []
+    for follow_to_fetch in follow_ops_to_fetch:
+        follower = follow_to_fetch[0]
+        # follows does not need entity type in follow_to_fetch[1]
+        followee = follow_to_fetch[2]
+        and_queries.append(
+            and_(
+                Follow.followee_user_id == followee,
+                Follow.follower_user_id == follower,
+                Follow.is_current == True,
+            )
+        )
+    follows: List[Follow] = session.query(Follow).filter(or_(*and_queries)).all()
+    existing_entities[EntityType.FOLLOW] = {
+        get_record_key(
+            follow.follower_user_id, EntityType.USER, follow.followee_user_id
+        ): follow
+        for follow in follows
+    }
+
+    # SAVES
+    saves_to_fetch: Set[Tuple] = entities_to_fetch[EntityType.SAVE]
+    and_queries = []
+    for save_to_fetch in saves_to_fetch:
+        user_id = save_to_fetch[0]
+        entity_type = save_to_fetch[1]
+        entity_id = save_to_fetch[2]
+        and_queries.append(
+            and_(
+                Save.user_id == user_id,
+                Save.save_type == entity_type.lower(),
+                Save.save_item_id == entity_id,
+                Save.is_current == True,
+            )
+        )
+    saves: List[Save] = session.query(Save).filter(or_(*and_queries)).all()
+    existing_entities[EntityType.SAVE] = {
+        get_record_key(save.user_id, save.save_type, save.save_item_id): save
+        for save in saves
+    }
+
+    # REPOSTS
+    reposts_to_fetch: Set[Tuple] = entities_to_fetch[EntityType.REPOST]
+    and_queries = []
+    for repost_to_fetch in reposts_to_fetch:
+        user_id = repost_to_fetch[0]
+        entity_type = repost_to_fetch[1]
+        entity_id = repost_to_fetch[2]
+        and_queries.append(
+            and_(
+                Repost.user_id == user_id,
+                Repost.repost_type == entity_type.lower(),
+                Repost.repost_item_id == entity_id,
+                Repost.is_current == True,
+            )
+        )
+    reposts: List[Repost] = session.query(Repost).filter(or_(*and_queries)).all()
```

Comment:
for repost, save, and follows - if the and_queries is an empty list, will it try to fetch everything? 
ie. the query would end up as `select * from reposts` and take a long time to finish because there's so much?